### PR TITLE
Detect package root path from language repo when creating work item

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Package root path is not available when creating a package work item from release planner. Thich change is to automatically detect the package root path from language repo by the pipeline, so it won't be empty for any package work items created from release planner.

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
Fixes https://github.com/Azure/azure-sdk-tools/issues/7851

File: eng/scripts/Create-Package-WorkItem.ps1 [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        "C" { return "azure-sdk-for-c" }
+        "C++" { return "azure-sdk-for-cpp" }
+    }
+    Write-Host "Unknown language to map it to repo name. Language: [$language]"
+    return ""
+}
+
+function Clone-Repo($repoName, $targetDir)
+{
+    git clone "https://github.com/azure/$repoName.git"  $targetDir
```
Comment (weshaggard) [inline eng/scripts/Create-Package-WorkItem.ps1:44]: Might not be necessary but we should consider a shallow clone to speed things up.

File: eng/scripts/Create-Package-WorkItem.ps1 [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        exit 1
+    }
+    $clonedRepoPath = Join-Path $WorkingDir $repoName
+    Clone-Repo $repoName $clonedRepoPath
+    
+    # load common script helpers from cloned language repo
+    . (Join-Path $clonedRepoPath "eng/common/scripts/common.ps1")
+    Write-Host "Language repo root path: [$RepoRoot]"
+    # Parse package properties from cloned repo and find the package repo path
+    $packageProperties = Get-PkgProperties -PackageName $PackageName -ServiceDirectory ""
```
Comment (weshaggard) [inline eng/scripts/Create-Package-WorkItem.ps1:69]:
```suggestion
    $packageProperties = Get-PkgProperties -PackageName $PackageName
```

File: eng/scripts/Create-Package-WorkItem.ps1 [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+Write-Host "Package Root Path: [$PackageRootPath]"
+# Create or update package work item  
+&$EngCommonScriptsDir/Update-DevOps-Release-WorkItem.ps1 `
+    -language $Language `
+    -packageName $PackageName `
+    -version $PackageVersion `
+    -plannedDate $ReleaseDate `
+    -packageRepoPath $PackageRootPath `
+    -packageType $PackageType `
```
Comment (weshaggard) [inline eng/scripts/Create-Package-WorkItem.ps1:93]: What about `-packageNewLibrary`? https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Prepare-Release.ps1#L149

Should we use more of the package metadata we get from Get-PkgProperties? I also wonder if we should potentially validate those values with what you passing to pipeline.

**Comment by weshaggard** [comment]
What is your goal with these changes? Is this only for temporary purposes as we talked about deprecating this pipeline.

File: eng/scripts/Create-Package-WorkItem.ps1 (hunk 1 above)
Comment (praveenkuttappan) [reply eng/scripts/Create-Package-WorkItem.ps1:44]: We don't know the service directory to sparse checkout only the package path. Are you suggesting to skip any test directories? I can update the depth to make it as shallow clone.

**Comment by praveenkuttappan** [comment]
> What is your goal with these changes? Is this only for temporary purposes as we talked about deprecating this pipeline.

Correct. It's only temporary work around until we remove the need to create package work item from release planner.

File: eng/scripts/Create-Package-WorkItem.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Create-Package-WorkItem.ps1]: I was just meaning to we could do a sparse clone which doesn't include any history which reduces the cost of cloning. Either way it is just an optimization.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**